### PR TITLE
Upgrading versions to remove CVE-2021-42550, CVE-2021-42340, CVE-2022-23181 ...

### PR DESCRIPTION
**Pull request by dutta1kartik3**

### CODE REVIEW

**Comment by dutta1kartik3** [comment]
All checks are passing ....
https://github.com/dutta1kartik3/gretty/pull/3/checks

**Comment by dutta1kartik3** [comment]
@f4lco @boris-petrov We need to upgrade versions of logback to avert BDSA-2021-3818, Apache Tomcat to avert BDSA-2021-3085, BDSA-2022-0275, Jetty to avert BDSA-2021-2098

**Comment by boris-petrov** [comment]
@dutta1kartik3 - thanks for the PR! I'm waiting on @f4lco to take a look and then we can merge. By the way, Tomcat tagged [10.0.17](https://github.com/apache/tomcat/tree/10.0.17) so I guess they'll release it any time now, you might want to update the PR when they do.

File: gradle.properties [hunk 1]
```diff
@@ -7,17 +7,17 @@ developerName = Andrey Hihlovskiy
...
 version = 4.0.1-SNAPSHOT
-jetty11_version = 11.0.7
+jetty11_version = 11.0.8
 jetty11_servlet_api_version = 5.0.0
-tomcat10_version = 10.0.14
+tomcat10_version = 10.0.16
 tomcat10_servlet_api_version = 5.0.0
 asm_version = 9.2
 spock_version = 1.3-groovy-2.5
 # Updating Logback to 1.2.3 causes a build failure: https://travis-ci.org/github/gretty-gradle-plugin/gretty/jobs/723685603
```
Comment (boris-petrov) [inline gradle.properties:16]: Also, please remove this comment.

**Comment by f4lco** [comment]
Thank you for looking into this, largely this patch looks very good. Please fix the minor remarks of @boris-petrov.

However, we'll need to do #249, and remove the ability to configure logging in Gretty via Groovy, because
1. Logback dropped configuration via Groovy script in 1.2.9, and Gretty will not receive full mitigation of the linked vulnerabilities if we still allow it
2. I do not want to place the burden of maintaining Logback configuration via Groovy on the Gretty maintainers. That feature is gone upstream, and so it disappears from Gretty, too.

So we can drop `GafferConfiguratorEx`. That class is not used anyway, see the following comment:

https://github.com/gretty-gradle-plugin/gretty/blob/9cf7cbb7a2c059449ab9e1a7687826d9702ecee3/libs/gretty-runner/src/main/groovy/org/akhikhl/gretty/Runner.groovy#L83-L95

So the baseline for this PR is removing `GafferConfiguratorEx`, which is easy, because the class has no call sites.

I do not want to place an excessive burden on you, but please consider doing the following steps from https://github.com/gretty-gradle-plugin/gretty/issues/249#issuecomment-1048943276as your time & willingness permit. Those steps include reworking the logic inside `Runner`, porting the default logging configuration, and cleaning up the integration tests.

Many thanks, and if you run in any problems or have questions, please hit us up on Gitter / Matrix.

**Comment by dutta1kartik3** [comment]
@f4lco I have fixed the minor comment of @boris-petrov . I am wondering if its possible to merge this PR, create a snapshot build of gretty so that I can be unblocked on my own project and then work on the issues you have listed on #249 as part of a separate PR ? I'll be happy to help out with that in a separate PR. What do you think?

**Comment by f4lco** [comment]
That would be totally possible and greatly appreciated 🙏

I would just have to coordinate a patch release with @boris-petrov. ETA might be next week.

**Comment by dutta1kartik3** [comment]
> That would be totally possible and greatly appreciated 🙏 I would just have to coordinate a patch release with @boris-petrov. ETA might be next week.

Thanks a lot, @f4lco . Really appreciate it.

**Comment by boris-petrov** [comment]
@f4lco - if you think this is ready to be merged, please do and I'll cut a new release.

@dutta1kartik3 - you need a new release of Gretty 3 or 4? This PR currently targets Gretty 4.

**Comment by dutta1kartik3** [comment]
@boris-petrov new release of Gretty 4

**Comment by boris-petrov** [comment]
@dutta1kartik3 - 4.0.1 is out!

**Comment by dutta1kartik3** [comment]
Thank you @boris-petrov

**Pull request merged**